### PR TITLE
[Snaps E2E] Remove / Replace superflous arbitrary delays in snaps E2E tests

### PR DESCRIPTION
**Pull request by bowensanders**
## **Description**

This PR is intended to reduce the amount of CI time used by the snaps E2E tests by removing or replacing as many delays as possible without having a negative impact on test flakiness.

Fixes: 76 1000ms delays removed
24 500ms delays removed

**old test-e2e-chrome-flask speeds** (using v 11.10 as benchmark) -
shortest snaps test - 1m29s
longest snaps test - 2m23s
longest test in run - 3m16s (longest test is from accounts)
(see [this page](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/68846/workflows/68299190-7158-48d3-9527-45b702c01f19/jobs/2285255) for details)

**new test-e2e-chrome-flask speeds**
shortest snaps test - 1m10s
longest snaps test - 1m52
(see [this page](https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/68812/workflows/9382b2e8-a997-430e-8bd9-20b900bcea2f/jobs/2283573) for details) (edited)

## **Manual testing steps**

1. this only effects automated CI tests

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by FrederikBolding — COMMENTED** [review]
Please delete all the commented out lines before we merge this

**Comment by bowensanders** [comment]
comments removed

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js [hunk 1]
```diff
@@ -64,7 +68,7 @@ describe('Test Snap TxInsights-v2', function () {
         // switch to test-snaps page and get accounts
         await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
         await driver.clickElement('#getAccounts');
-        await driver.delay(1000);
+        // await driver.delay(1000);
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-txinsights-v2.spec.js:71]:
```suggestion
```

File: test/e2e/snaps/test-snap-txinsights-v2.spec.js [hunk 2]
```diff
@@ -86,15 +93,18 @@ describe('Test Snap TxInsights-v2', function () {
         windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
         await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
         await driver.clickElement('#sendInsights');
-        await driver.delay(1000);
+        // await driver.delay(1000);
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-txinsights-v2.spec.js:96]:
```suggestion
```

**Review by FrederikBolding — COMMENTED** [review]
Spotted a few more

**Review by FrederikBolding — APPROVED** [review]

**Review by david0xd — APPROVED** [review]

**Pull request merged**